lib/spec.go

```diff
@@ -44,6 +45,40 @@ type Mock struct {
 	calls int
 }
 
+// Mocks metadata such as workspace
+type MockMetadata struct {
+	Workspace string
+}
+
+type assert struct {
+	Type      string         `hcl:"type,label"`
+	Name      string         `hcl:"name,label"`
+	Config    hcl.Body       `hcl:",remain"`
+	DependsOn hcl.Expression `hcl:"depends_on,attr"`
+}
+
+type mock struct {
+	Type   string   `hcl:"type,label"`
+	Name   string   `hcl:"name,label"`
+	Config hcl.Body `hcl:",remain"`
+}
+
+type mock_metadata struct {
+	Config hcl.Body `hcl:",remain"`
+}
+
+type MetadataConfig struct {
+	Workspace string `hcl:"workspace"`
+}
+
+type root struct {
+	Asserts      []*assert        `hcl:"assert,block"`
+	Refutes      []*assert        `hcl:"refute,block"`
+	Mocks        []*mock          `hcl:"mock,block"`
+	MockMetadata []*mock_metadata `hcl:"mock_metadata,block"`
```

Comment:
```suggestion
	MockMetadata *mock_metadata `hcl:"mock_metadata,block"`
```
You don't expect more than 1 MockMetadata per spec

Reply:
I was worried about making it required by accident. Not sure how to do the with hcl.